### ISSUE TITLE
Error running double pendulum animation example

### ISSUE THREAD

**Issue by jakirkham** [matplotlib/matplotlib#4901]
Using this example ( http://matplotlib.org/examples/animation/double_pendulum_animated.html ) in an iPython Notebook on Mac OS 10.9 and version 1.4.3 of matplotlib. I encounter infinite recursion. See notebook here ( http://nbviewer.ipython.org/github/jakirkham/double_pendulum_animated/blob/master/double_pendulum_animated.ipynb ).

**Comment by tacaswell** [comment]
what version of mpl and what backend?

**Comment by jakirkham** [comment]
Matplotlib version was 1.4.3. Backend is whatever the default is. This is using the matplotlib from conda.

**Comment by dopplershift** [comment]
Any chance you were running master?

**Comment by jenshnielsen** [comment]
Looking at the Traceback it must be the OSX backend

**Comment by jenshnielsen** [comment]
```
AttributeError: 'FigureCanvasMac' object has no attribute 'restore_region'
```

**Comment by dopplershift** [comment]
I can reproduce this just from the command line using OSX backend; works with Qt4Agg backend.

But is this just a manifestation of the fact that the OSX backend doesn't support blitting (at least like the other backends do)?

**Comment by dopplershift** [comment]
Works if I change call to animation to set `blit=False`.

**Comment by jakirkham** [comment]
@dopplershift, nope, wasn't using master.

**Comment by mdehoon** [comment]
Same issue as #531

**Comment by mdehoon** [comment]
Using the tkagg backend, with the current master, the script also encounters into an error when blitting:
```
...
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/matplotlib-1.5.dev1-py2.7-macosx-10.9-x86_64.egg/matplotlib/animation.py", line 862, in _blit_clear
    a.figure.canvas.restore_region(bg_cache[a])
KeyError: <matplotlib.axes._subplots.AxesSubplot object at 0x10cc9bc10>
```

The script works correctly if blit=False.

**Comment by tacaswell** [comment]
I think this is reported in the that issue with a list of 1.5 bugs (but
with less detail there so i am not sure).

**Comment by tacaswell** [comment]
@mdehoon I can reproduce your bug only if interactive mode in on.

Interestingly if you resize the window to force a re-draw the animation starts up so I suspect that this is due to something forcing a draw too soon (which is probably the PR I put in to fix inline).

if you start the script with `python -i animation.py` it runs very slowly on current master due to the stale flags being agressively processed (so lots of extra draws), but it works correctly with `ipython -i animation.py` because the `draw_idle` calls are triggered by an IPython event hook which does not get hit by this code path.  One of the commits in #4738 makes this a bit better, but if you switch to that branch you can call `plt.ion()` and `plt.ioff()` and watch the frame rate change.

**Comment by tacaswell** [comment]
@mdehoon I have pushed commits to #4738 that seems to fix all of these issues.

**Comment by mdboom** [comment]
Closed by #4738
